objects.js

```diff
@@ -2640,87 +2640,137 @@ SpriteMorph.prototype.reporterize = function (expressionString) {
     // highly experimental Christmas Easter Egg 2016 :-)
     var ast;
 
-    function parseInfix(expression, operator, already) {
-        // very basic binary infix parser for arithmetic expressions
-        // with strict left-to-right operator precedence (like in Smalltalk)
-        // which can be overriden by - nested - parentheses.
-        // assumes well-formed expressions, no graceful error handling yet.
-
-        var inputs = ['', ''],
-            idx = 0,
-            ch;
-
-        function format(value) {
-            return value instanceof Array || isNaN(+value) ? value : +value;
-        }
-
-        function nested() {
-            var level = 1,
-                expr = '';
-            while (idx < expression.length) {
-                ch = expression[idx];
-                idx += 1;
-                switch (ch) {
-                case '(':
-                    level += 1;
-                    break;
-                case ')':
-                    level -= 1;
-                    if (!level) {
-                        return expr;
-                    }
-                    break;
-                }
-                expr += ch;
+    function tokenize(expressionString) {
+        var tokens = [];
+        var token;
+        expressionString = expressionString.trim();
+        function findIdent() {
+            var next = "";
+            while (expressionString.length > 0 && /[A-Za-z_0-9]/.test(expressionString[0])) {
+                next += expressionString[0];
+                expressionString = expressionString.slice(1);
+            }
+            return next;
+        }
+        function findNumber() {
+            var next = "";
+            while (expressionString.length > 0 && /\d|\./.test(expressionString[0])) {
+                next += expressionString[0];
+                expressionString = expressionString.slice(1);
+            }
+            return parseFloat(next);
+        }
+        while (expressionString.length > 0) {
+            expressionString = expressionString.trim();
+            if (/[A-Za-z_]/.test(expressionString[0])) {
```

Comment:
What it _is_ doing and what it is _trying_ do are two separate things. The reason to name things is because they help convey programmer intent, not just function. :) When I look at this I question whether or not `-` can be in a variable name or whether this is a complete set of characters. I completely forgot to ask that question before....